src/main.cpp

```diff
@@ -2080,6 +2081,50 @@ void PartitionCheck(bool (*initialDownloadCheck)(), CCriticalSection& cs, const
     }
 }
 
+// Protected by cs_main
+static VersionBitsCache versionbitscache;
+
+int32_t ComputeBlockVersion(const CBlockIndex* pindexPrev, const Consensus::Params& params)
```

Comment:
I think it'd be nice if we had test coverage for this function.  I believe that as it stands, if this code were busted, then I'm not sure any tests would necessarily fail?

So far the best idea I have is that when we deploy a new soft fork, we write a unit test that exercises this function and verifies the right bit will be set at the right time (using a fake mainnet chain like the one you built in the versionbits unit test). 

I think we should also have regtest RPC tests which exercise proposed soft forks as well, but I can't think of any way to write such tests to adequately test mainnet parameters...